src/TuxxState.js

```diff
@@ -0,0 +1,338 @@
+// @param input UNKNOWN: value we are performing invariant check upon
+var invariantArrayCheck = function (input) {
+  var isArray = Array.isArray(input);
+  invariant(isArray, 'Cannot perform operation on "%s" because it is not an array.', input);
+};
+
+
+// invariantValueCheck FUNCTION: internal function that ensures a given value is not stored in an array or an object
+// @param input UNKNOWN: value we are performing invariant check upon
+var invariantValueCheck = function (input) {
+  var isObject = typeof(input) === "object";
+  invariant(!isObject, 'Cannot perform operation on "%s" because it must not be an array or object.', input);
+};
+
+// invariantNumberOrStringCheck FUNCTION: internal function that ensures a given value is a number or string or throws an error
+// @param input UNKNOWN: value we are performing invariant check upon
+var invariantNumberOrStringCheck = function (input) {
+  var isNumberOrString = typeof(input) === "number" || typeof(input) === "string";
+  invariant(isNumberOrString, 'Cannot perform operation on "%s" because it is not of type number or of type string.', input);
+};
+
+// invariantArgCheck FUNCTION: internal function that ensures a given value is defined and is an object or throws an error
+// @param input UNKNOWN: value we are performing invariant check upon
+var invariantArgCheck = function (input) {
+  var isObjectArgument = Object.prototype.toString.call(input) === '[object Object]';
+  invariant(isObjectArgument, 'This function requires an object as an argument.');
+};
+
+// mixin to React class that will provide convenience methods for updating state
+module.exports = {
+
+  // addState FUNCTION: adds the values of the deepest keys in the passed in object to the corresponding deepest keys in the current state or throws an error if keys don't match
+  // @param propsToAdd OBJECT: required object argument where the deepest keys are numbers or strings
+  // @param callback FUNCTION: optional callback argument that will be executed once setState is completed and the component is re-rendered
+  addState: function (propsToAdd, callback) {
+    // throw error if propsToAdd is undefined or not an object
+    invariantArgCheck(propsToAdd);
+    // build new state object
+    var newState = buildNewState(this.state, propsToAdd, function (newState, key, currentStateAtKey, newPropsAtKey) {
+      // throw error if newPropsAtKey is not a number or string
+      invariantNumberOrStringCheck(newPropsAtKey);
+      // perform addition on the corresponding deep keys
+      newState[key] = currentStateAtKey + newPropsAtKey;
+    });
+    // update state with new values and pass callback (if provided), triggering re-render
+    this.setState(newState, callback);
+  },
+
+  // performArithmeticNumberOperations FUNCTION: internal function to perform some arithmetic number operation on the deepest keys in the passed in object from the deepest keys in the current state or throws an error if keys dont match
```

Comment:
This internal function should be moved outside the object that is being exported.